### DISCUSSION TITLE
Meeting minutes

### DISCUSSION THREAD

**Discussion by JovieL25** [smiths/aorta#11] [General]
September 26, 2022

1. Work on https://github.com/smiths/aorta/issues/2 to include the existing modules. For each module, include its service, secret (likely change in the module), and typical use case of each module.
2. Add regression test https://github.com/smiths/aorta/issues/5, we can use a small set of the full sample to test.
3. MIS - we need to see if it fits the system.

**Comment by JovieL25** [comment]
October 3, 2022

1. Continue to work on #2, especially on adding the anticipated changes, because the modules should be decomposed by the anticipated change.
2. The inputs should be kept in one module so that we can change them globally at any time during the process.
3. Code inspection when we are comfortable (comment: Code inspection is to get approval on the algorithm from domain expert?)
4. Display module and input module can be formed into one. 

The next meeting is on October 19th, 13:30.

**Comment by smiths** [comment]
@JovieL25, in response to your comment on your point 3, the purpose of the code inspection is to provide evidence that the code is trustworthy.  It isn't exactly the same as getting approval from the domain expert.  We take the domain expert through the code to try to find problems.  If we don't find any, that increase our confidence in the quality of the code.  If we do find problems, we fix them and then repeat the exercise.

We should also consider a code walkthrough, as opposed to a code inspection.  A code walkthrough would be a nice way to have a real conversation with the domain expert.

**Comment by JovieL25** [comment]
October 19, 2022


1. Investigate the software Paraview, particularly about reading .dicom files, the interaction of files, and Python script to interact with image objects. (#15)

2. Investigate the module decomposition for Aorta segmentation. (#14)

The next meeting is on October 30th, at 13:30

**Comment by JovieL25** [comment]
November 2, 2022

1. Add instructions on how to set up 3D slicer. I will include the steps to load .dicom files, use the Markups module to select areas of interest, and load extensions and modules.
2. Upload my extensions in GitHub (After organizing the modules).

The next meeting is on November 9th, at 13:30

**Comment by JovieL25** [comment]
November 9, 2022

1. Need to modify current UI to step specific (Select crop area, select seeds, result)

The next meeting is on November 16th, at 13:30

**Comment by JovieL25** [comment]
November 23, 2022

1. We can now use the "volume rendering" module from 3D Slicer to make the first crop of Aorta parts. By rendering the volume, we can see clearly a 3D geometry and we can control the crop region while verifying the output.
2. Investigate finding new seeds value for the same algorithm. The difficult part is working out the numbers with the coordinate system and the corresponding indexes.
3. Upload the instructions to use Volume rendering to crop Aorta.

The next meeting is on November 30th, at 13:30.

**Comment by JovieL25** [comment]
November 30, 2022

1. Work on the rest of the segmentation algorithm for the extension to work exactly as it was as a script
2. Improve and clarify the current instructions to crop volume and get the segmentation working.

The next meeting is on December 19th, at 13:30.

**Comment by JovieL25** [comment]
December 21, 2022

As part of the next phase, gathering pieces of evidence for the assurance case of the current software:
1. Create known test cases for the testing. First I will look up ways to draw fake straight tubes, then add some noises. Or create a physical phantom and get the CT images. Both should provide a known result that we can compare with the result of the algorithm.
2. We will perform a code inspection as part of the evidence.
3. Start working on the report.

The next meeting is on January 9th, 2023, at 13:30.

**Reply by smiths** [reply]
Sounds good @JovieL25.  We will also look to test against manually created solution for real images.  We'll need to figure out what it means to compare to images.  The bar isn't to have a pixel by pixel comparison, since that will only work for simple images.  We may want to compare using the dice coefficient.  Another option is to compare using derived statistics, like the length of the aorta and diameter of the aorta.

To help you start working on the report, here are some resources on writing:

https://gitlab.cas.mcmaster.ca/SEforSC/se4sc/-/blob/git-svn/GradStudResources/README.txt

In particular you might want to follow [Geneva's LaTeX template](https://gitlab.cas.mcmaster.ca/SEforSC/se4sc/-/tree/git-svn/GradStudResources/EmptyThesis/GenevaLaTeXTemplates) and look at the sample [MEng reports.](https://gitlab.cas.mcmaster.ca/SEforSC/se4sc/-/blob/git-svn/GradStudResources/ExampleMEngReports.txt)

**Comment by JovieL25** [comment]
January 16, 2023

1. Find some colleagues to follow the README instructions and get some feedback.
2. Use Projects to keep track of our progress.
3. Split the timeline into smaller issues and remove the proposed timeline issue.

The next meeting is on January 23rd, 2023, at 13:30.

**Reply by smiths** [reply]
As you create the issues that need to be addressed for your project, I will add some initial thoughts and "getting started" information to each.

**Comment by JovieL25** [comment]
January 23, 2023

1. Update README based on Smith's feedback.
2. Update the code to change the hard-coded values to some symbolic constants (improve code readability).
3. Ask Xuanming for test cases.

The next meeting is on January 30th, 2023, at 13:30.

**Comment by JovieL25** [comment]
January 30, 2023

1. Work on the instruction videos
2. Work on the continuous integration tests infrastructure, and research on getting some metrics of the volume.
3. Research methods to compare volumes

The next meeting is on February 6th, 2023, at 13:30.

**Comment by JovieL25** [comment]
February 6, 2023

1. Instruction videos can be part of the usability evidence.
2. Implement the Sorensen-Dice coefficient for comparing images
3. Refactor the segmentation algorithm to the original code (improve it over time).
4. Work on Software Requirement specification.
5. (Optional) Add a robot voiceover to the video.

The next meeting is on February 13th, 2023, at 13:30.

**Comment by JovieL25** [comment]
February 13, 2023

1. Improve README based on Dr. Smith's comments
2. Add an overview including conventions, and terminology in design documents.
3. Add the story of finding the bugs with continuous integration testing to the final report.
4. Working on SRS, mainly on GS and System description, going to borrow some parts from [Mojdeh's SRS](https://gitlab.cas.mcmaster.ca/SEforSC/se4sc/-/tree/git-svn/GradStudents/Mojdeh/Documents)

The next meeting is on February 20th, 2023, at 13:30.

**Comment by JovieL25** [comment]
February 28, 2023

1. Change the minimum values on the GUI.
2. Keep sagittal segmentation optional.

The next meeting is on March 6th, 2023, at 13:30.

**Comment by JovieL25** [comment]
March 06, 2023

1. Work on SRS on the following part: 
    a. Data definition: add a definition for the image (a sequence of 2d images/slices)
    b. Assumptions (e.g. User provides CT scans of the chest which include aorta)
    c. Instance model for the region of interest, and segmentation
    d. requirements, non-functional requirements
    After I have worked on these items, I will add an issue to let @smiths review.

2. Add a pseudocode in the design document which explains that the algorithm works slice by slice with a loop.

The next meeting is on March 13th, 2023, at 13:30.

**Reply by smiths** [reply]
Great summary.  Thank you @JovieL25.

**Comment by JovieL25** [comment]
March 21, 2023
1. Work on the design document
   * Add description for some of the helper functions
   * Clarify pseudocode, assuming that users have not used Python and Numpy.
   * Fix typo
2. SRS
   * Add voxel dd, modify IM with math notation
   * Add seeds value in requirements
3. Work on a list of items for a code walkthrough
   * Links to read before the walkthrough
   * GitHub, source code, and design document.
The next meeting is on March 27th, 2023, at 13:30.

**Comment by JovieL25** [comment]
March 28, 2023
1. Work on the design document
   * Add pictures for each of the steps described in the algorithm description
   * Simplify source codes if possible
2. Work on a list of items for a code walkthrough
   * Demo with 3D Slicer
   * SITK documentations
   * Other links to read before the walkthrough
   * GitHub, source code, and design document.
4. Work on SRS as mentioned on March 21 meeting's minutes.

The next meeting is on April 3rd, 2023, at 13:30.

**Comment by JovieL25** [comment]
April 3, 2023

1. We will keep going our code walkthrough from where it [left off](https://github.com/smiths/aorta/blob/design-document/src/SlicerExtension/AortaGeometryReconstructor/AortaGeomReconDisplayModule/AortaGeomReconDisplayModuleLib/AortaSegmenter.py#L261)
2. Work on SRS as mentioned on March 21 meeting's minutes.

The next meeting is on April 10th, 2023, at 13:30.

**Comment by JovieL25** [comment]
April 20, 2023

1. Work on SRS.
2. Advice for writing in our next meeting.
3. Prepare some materials for the code walkthrough.

The next meeting is on April 24th, 2023, at 13:30.

**Comment by JovieL25** [comment]
April 24, 2023

1. Work on https://github.com/smiths/aorta/issues/38.
2. Prepare links for the code walkthrough at 11 AM.
3. Work on https://github.com/smiths/aorta/issues/6
4. Work on https://github.com/smiths/aorta/issues/8
5. Start a draft of the final report, I should start with ToC.

The meeting on April 25th for a code walkthrough is at 11 AM.
The usual meeting is on May 1st at 13:30.

**Comment by JovieL25** [comment]
May 1st, 2023

1. Work on https://github.com/smiths/aorta/issues/6, aiming to finish this week.
2. Work on https://github.com/smiths/aorta/issues/8.
3. Thought on changing the code walkthrough to an algorithm review to get more insights or build confidence in our solution.

The next meeting is on May 8th, 2023, at 13:30.

**Comment by JovieL25** [comment]
May 8, 2023

1. Work on implementing K-means clustering with the current segmentation algorithm.
2. Look up the modules in the MG and find the corresponding code.
3. Work on https://github.com/smiths/aorta/issues/6
4. Work on https://github.com/smiths/aorta/issues/8.

(Don't forget about MBTI [test ](https://www.16personalities.com/free-personality-test)) (I am ENFP)

The next meeting is on May 15th, 2023, at 13:30.

**Reply by smiths** [reply]
Sounds good @JovieL25.  I'm an advocate (INFJ-T).  :smile:

**Comment by JovieL25** [comment]
May 15, 2023

1. Prepare questions for Wednesday's meeting.
2. [Link](http://insightsoftwareconsortium.github.io/SimpleITK-Notebooks/Python_html/31_Levelset_Segmentation.html) to Level Sets segmentation tutorial, this and Kmean cluster should be mentioned for an appeal to authority.

The next meeting is on May 17th, 2023, at 13.

**Reply by smiths** [reply]
Looks good.  @JovieL25, can you also provide the link to the tutorial that inspired the original version of aorta?

**Reply by JovieL25** [reply]
Yes, the link to Level Sets segmentation is the tutorial that inspired the original segmentation algorithm,

**Reply by smiths** [reply]
Ooops, I missed the link!  Thank you!
